dig.go

```diff
@@ -111,6 +111,25 @@ type InvokeOption interface {
 	unimplemented()
 }
 
+// A dotNode represents a node in a Container's DOT-format graph.
+type dotNode struct {
+	Type     string
+	name     string
+	optional bool
+	group    string
+}
+
+// A dotEdge represents an edge in a Container's DOT-format graph.
+type dotEdge struct {
+	param  dotNode
+	result dotNode
+}
+
+// A dotGraph represents the DOT-format graph in a Container.
+type dotGraph struct {
+	edges []dotEdge
```

Comment:
Seeing as we might need to mutate the `dotEdge` contents (i.e. in the presence of `Invoke` failures), let's use pointers here.

```
type dotGraph struct {
  edges []*dotEdge
}
```

This would also apply to the above:
```
type dotEdge struct {
  param *dotNode
  result *dotNode
}
```